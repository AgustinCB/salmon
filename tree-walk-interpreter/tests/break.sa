var i = 0;
print "CACA";
while (i < 5) {
  print "HOLA";
  break;
  i = i + 1;
}

print "CHAU";

for (var i = 0; i < 5; i = i + 1) {
  print "HOLA0";
  break;
}

print "CHAU0";

for (var i = 0; i < 5; i = i + 1) {
  print "HOLA1";
  if (i == 1) break;
  print "HOLA1.1";
}

print "CHAU1";

for (var i = 0; i < 5; i = i + 1) {
  print "HOLA2";
  for (var i = 0; i < 5; i = i + 1) {
    print "INNER HOLA2";
    break;
  }
  print "AFTER INNER HOLA2";
  break;
  print "DO NOT SEE2";
}

print "CHAU2";

for (var i = 0; i < 5; i = i + 1) {
  print "HOLA3";
  for (var i = 0; i < 5; i = i + 1) {
    print "INNER HOLA3";
    if (i == 1) break;
    print "SEE ONCE3";
  }
  print "AFTER INNER HOLA3";
  break;
  print "DO NOT SEE3";
}

print "CHAU3";
