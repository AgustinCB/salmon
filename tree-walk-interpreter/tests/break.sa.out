CACA
HOLA
CHAU
HOLA0
CHAU0
HOLA1
HOLA1.1
HOLA1
CHAU1
HOLA2
INNER HOLA2
AFTER INNER HOLA2
CHAU2
HOLA3
INNER HOLA3
SEE ONCE3
INNER HOLA3
AFTER INNER HOLA3
CHAU3
